packages/gatsby-plugin-fastify/src/plugins/server-routes.ts

```diff
@@ -105,20 +101,13 @@ export const handleServerRoutes: FastifyPluginAsync<{
               req: request,
             });
 
-            const results = await renderHTML({ data: pageQueryData });
-
-            if (page.mode === `SSR`) {
-              if (pageQueryData?.serverDataHeaders) {
-                reply.headers(pageQueryData.serverDataHeaders);
-              }
+            // add pageQueryData to reply so it can be accessed in the onSend hook
+            reply.pageQueryData = pageQueryData;
 
-              if (pageQueryData?.serverDataStatus) {
-                reply.code(pageQueryData.serverDataStatus);
-              }
-            }
+            const results = await renderHTML({ data: pageQueryData });
 
-            if (page.mode === "DSG") {
```

Comment:
This may have been a mistake previously as gatsby hosting does set the `NEVER_CACHE_HEADER` on SSR routes as [seen above](https://github.com/gatsby-uc/plugins/pull/392#discussion_r1135656503)